Add re-vote maps on tie

thboss opened the pull request.

thboss committed:
Add re-vote maps on tie

thboss committed:
flake8

cameronshinn commented:
I'm assuming you have all the `map_pool` function arguments added in because I said that we don't want to modify `self.map_pool` to keep it the same as the guild's value in the database.

Assuming that is the case I think it would just be easier to modify `self.map_pool` and take out the function arguments. However, if there is any other reason we need to preserve the original map pool then we could make a new attribute to store the map pool as it decreases. Could call it `self.runoff_map_pool` perhaps?

thboss commented:
@cameronshinn I didn't understand the problem of that, modifying `self.map_pool` in `MapVoteMenu` class doesn't change the map pool in the database!

cameronshinn commented:
Yes it doesn't. There isn't any problem if you want to modify `self.map_pool`. I'd prefer that than have to add an argument to every single function.

thboss committed:
Improve volunteer pick method

- Fix user not in list error when a player in team one trying to be a volunteer for team two
- Block volunteers to pick themselves because that's will broke picking order

cameronshinn commented:
@thboss can you move the volunteer pick improvements to another PR? Also are you going to remove the `map_pool` arguments that you added? You can just use `self.map_pool` instead of adding an argument to every method.

thboss committed:
Revert "Improve volunteer pick method"

This reverts commit bfa6f9d35de0ecb0fc16dd0057d4dca3cd73c626.

thboss committed:
Remove map_pool argument from map_method functions

thboss committed:
flake8

B3none commented:
It makes sense to have these changes separate @thboss - would be good to see this stuff merged :)

thboss commented:
I can't open multiple PRs

B3none commented:
@thboss you can

thboss committed:
Merge branch 'develop' of https://github.com/thboss/csgo-league-bot-map-votes-counter into develop

Commit:
Merge branch 'develop' into develop
